transformer_engine/pytorch/attention.py

```diff
@@ -2643,6 +2646,28 @@ def forward(
         if qkv_format is None:
             qkv_format = self.qkv_format
 
+        if inference_params is not None:
+            assert qkv_format == "sbhd"
+            assert self.layer_number is not None
```

Comment:
We should have a proper error message here telling the user to set the layer number.